### PR TITLE
NMS-9072: XmlCollector use parameters defined in the service definition in xml-source url

### PR DESCRIPTION
**Pull request by roskens**
NMS-9072: Pass the service parameters through and allow them to be used in the URLs and Request headers/data.

* JIRA: http://issues.opennms.org/browse/NMS-9072

### CODE REVIEW

**Comment by mvrueden** [comment]
This was originally targeted to be merged to master. Master should be protected and is for releases only. I changed the branch to be develop instead. Thank you (-:

File: protocols/xml/src/main/java/org/opennms/protocols/xml/collector/Sftp3gppXmlCollectionHandler.java [hunk 1]
```diff
@@ -76,6 +78,20 @@ public XmlCollectionSet collect(CollectionAgent agent, XmlDataCollection collect
...
+        for(Map.Entry<String, Object> entry : parameters.entrySet()) {
+            if(entry.getKey().equals("SERVICE"))
+                continue;
+            if(entry.getKey().equals("collection"))
+                continue;
+            if(entry.getKey().equals("xml-collection"))
+                continue;
+            if(entry.getKey().equals("handler-class"))
+                continue;
+            params.put(entry.getKey(), ParameterMap.getKeyedString(parameters, entry.getKey(), ""));
```
Comment (mvrueden) [inline protocols/xml/src/main/java/org/opennms/protocols/xml/collector/Sftp3gppXmlCollectionHandler.java]: This seems to be duplicated code, as it is used multiple times. Can we create a filter function or something like that? e.g:

``` java
// TODO add javadoc
// TODO verify that it actually works
public Map<String, String> filterParameters(Map<String, Object> input) {
   Set<String> blackList = Sets.newHashSet("SERVICE", "collection", "xml-collection", "handler-class");
   return   input.entrySet().stream()
                      .filter(e -> !blackList.contains(e.getKey())
                      .collect(Collectors.toMap(e -> e.getKey(), e -> ParameterMep.getKeyedString(input, e.getKey(), ""));
}
```

Comment (roskens) [inline protocols/xml/src/main/java/org/opennms/protocols/xml/collector/Sftp3gppXmlCollectionHandler.java]: Easy enough to do.

**Comment by roskens** [comment]
Rebased against develop, and squashed all commits into one.

**Comment by soleger** [comment]
I've rebased against foundation-2017 and pushed to a branch in our repo:

https://github.com/OpenNMS/opennms/tree/jira/NMS-9072

I'll merge after it passes Bamboo.

**Comment by soleger** [comment]
Merged the local branch, closing this PR.

**Pull request closed without merging**